src/datasets/builder.py

```diff
@@ -141,6 +141,7 @@ def create_config_id(
         self,
         config_kwargs: dict,
         custom_features: Optional[Features] = None,
+        fingerprint: Optional[str] = None,
```

Comment:
no need to modify create_config_id() imo, you can just use the user-defined config_id in the `__init__`

and in src/datasets/io/generator.py you can pass `Generator(..., config_id=fingerpring)`